usr/src/cmd/bhyve/bhyverun.c

```diff
@@ -1446,6 +1446,13 @@ main(int argc, char *argv[])
 #ifdef	__FreeBSD__
 	err = vm_setup_memory(ctx, memsize, VM_MMAP_ALL);
 #else
+	err = vm_arc_resv(ctx, memsize);
```

Comment:
Just to confirm, this is later freed by the vmm half of bhyve when the VM instance is destroyed? Want to make sure that a failed provision here doesn't leave memory reserved in the arc.

Reply:
Yeah, I saw that too.  My thinking was more so does a failed vmadm provision cause the /dev/vmm instance to be destroyed.  It's been awhile since I have looked at that code path.  If vmadm/vminfod time out on the provision I am not sure that the vmm instance is also fully cleaned up, potentially leaving things reserved.

Reply:
The amount reserved is reduced whenever the VMM allocated memory is released. How/when that memory is released by bhyve isn't changed.

(The reservation is done in two parts instead of one since the ARC releasing memory is asynchronous -- so I kept the loop/retry in the allocation part since it's still possible a VM w/ a lot of memory may still have to wait a few seconds before the ARC has finished releasing everything).

Reply:
I understand that this is tracked in the same lifecycle as the rest of the VMM stuff.  We just need to be aware that if the /dev/vmm device doesn't get cleaned up that it's accounted for and affects the arc size.

It's probably not an issue in practice.

Reply:
Yeah -- but it also means that the kernel memory is still allocated, so the ARC is already going to be limited anyway. This shouldn't substantially limit the ARC any further (since the ARC usage checks it does will already account for the same amount of VMM memory after it's allocated).